.github/workflows/codecov.yaml

```diff
@@ -0,0 +1,163 @@
+name: Code Coverage with codecov
+
+on:
+  schedule:
+    - cron: '25 */3 * * *'
+
+jobs:
+  codeocov-prep:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+
+  codecov:
+    runs-on: zephyr_runner
+    needs: codeocov-prep
+    container:
+      image: zephyrprojectrtos/ci:v0.18.4
+      options: '--entrypoint /bin/bash'
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: ["native_posix", "qemu_x86", "unit_testing"]
+    env:
+      ZEPHYR_SDK_INSTALL_DIR: /opt/toolchains/zephyr-sdk-0.13.1
+      CLANG_ROOT_DIR: /usr/lib/llvm-12
+    steps:
+      - name: Update PATH for west
+        run: |
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+
+      - name: checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: west setup
+        run: |
+          west init -l . || true
+          west update || ( rm -rf ../modules && west update)
```

Comment:
Why is this using rm -rf? It's safe to run west update twice in a row without it.

Reply:
oh, i had instances where one module failed to fetch correctly and was inconsistent, so west had issues with later runs because of the failing module.

Reply:
Hmm, that's weird. Do you have any logs or anything? West should be able to recover from a failed update, so this might be hiding a bug.

Reply:
@mbolivar-nordic I am not sure yet where the issue is, will remove this and wait for it to happen again I guess, to be able to look into what the issue is exactly and a better approach. thanks for flagging, was a workaround to move forward with this.

Reply:
> Do you have any logs or anything?

they are somewhere, but trying to find them now is going to be difficult :(

Reply:
Sure, thanks for that. We do retry `west update` in the current CI scripts to cover for network failures, but there shouldn't be a need to run `rm -rf` before doing that, at least when setting up a fresh workspace.